network-api/networkapi/minisites/models.py

```diff
@@ -0,0 +1,363 @@
+    height = blocks.IntegerBlock(default=450)
+
+    class Meta:
+        template = 'minisites/blocks/iframe_block.html'
+
+
+"""
+We'll need to figure out which components are truly "base" and
+which are bits that should be used in subclassing template-based
+page types.
+"""
+base_fields = [
+    ('heading', blocks.CharBlock()),
+    ('paragraph', blocks.RichTextBlock()),
+    ('image_text', ImageTextBlock()),
+    ('image', ImageChooserBlock()),
+    ('video', VideoBlock()),
+    ('iframe', iFrameBlock()),
+    ('linkbutton', LinkButtonBlock()),
+    ('spacer', VerticalSpacerBlock()),
+]
+
+
+class ModularPage(Page):
+    """
+    The base class offers universal component picking
+    """
+
+    header = models.CharField(
+        max_length=250,
+        blank=True
+    )
+
+    body = StreamField(base_fields)
+
+    content_panels = Page.content_panels + [
+        FieldPanel('header'),
+        StreamFieldPanel('body'),
+    ]
+
+    show_in_menus_default = True
+
+
+class MiniSiteNameSpace(ModularPage):
+    subpage_types = [
+        'CampaignPage',
+        'OpportunityPage',
+    ]
+
+    """
+    This is basically an abstract page type for setting up
+    minisite namespaces such as "campaign", "opportunity", etc.
+    """
+    def get_context(self, request):
+        """
+        Extend the context so that mini-site pages know what kind of tree
+        they live in, and what some of their local aspects are:
+        """
+        context = super(MiniSiteNameSpace, self).get_context(request)
+        ancestors = self.get_ancestors()
+        root = next((n for n in ancestors if n.specific_class == self.specific_class), self)
+        context['mini_site_title'] = root.title
+
+        children = self.get_children().live()
+        has_children = len(children) > 0
+        ancestors = self.get_ancestors()
+        root = next((root for root in ancestors if root.specific_class == self.specific_class), self)
+        context['root'] = root
+
+        is_top_page = (root == self)
+        context['is_top_page'] = is_top_page
+
+        context['singleton_page'] = (is_top_page and not has_children)
+
+        return context
+
+
+class CTA(SortableMixin):
+    title = models.CharField(
+        default='',
+        max_length=100,
+        help_text='Identify this component for other editors',
+    )
+
+    @property
+    def name(self):
+        return self.title
+
+    @name.setter
+    def name(self, value):
+        self.title = value
+
+    header = models.CharField(
+        max_length=500,
+        help_text='Heading that will display on page for this component',
+        blank=True
+    )
+
+    description = RichTextField(
+        help_text='Body (richtext) of component',
+        blank=True
+    )
+
+    newsletter = models.CharField(
+        max_length=100,
+        help_text='The (pre-existing) SalesForce newsletter to sign up for',
+        default='mozilla-foundation'
+    )
+
+    order = models.PositiveIntegerField(
```

Comment:
What's this for?

Reply:
A required field for `SortableMixin` snippets. Without an ordering field, you cannot reorder them.

Reply:
Is sorting something we already have for these/why do we need it?

Reply:
not sure, and excellent question. It's in there right now mostly "to keep our options open" but it's easy to take out before this PR is given the all clear if we don't see a usecase for having sorting available on CTA snippets.